regression/testing_tool.py

```diff
@@ -189,12 +201,23 @@ def test(self):
         stdout, stderr, rc = executor.run(test_case)
 
         if stdout == None:
+            if TestCase.SMT_ONLY:
+                return
             timeout_message ="\nTIMEOUT TEST: " + str(test_case.test_dir)
             self.fail(timeout_message)
-
+        RC_SMTLIB = rc == 3
+        SMT2_FILE_EXISTS = os.path.exists(f"{test_case.test_dir}.smt2")
+        if TestCase.SMT_ONLY and TestCase.SMT_SOLVER and RC_SMTLIB and SMT2_FILE_EXISTS:
+            # Try formula on SMT solver we are only interested in crashes
+            smt_command = [TestCase.SMT_SOLVER, f"{test_case.test_dir}.smt2"]
+            try:
+                p = subprocess.run(smt_command, stdout=PIPE, stderr=PIPE, timeout=10)
+                if p.returncode != 0:
+                    self.fail("Unable to parse the generated SMT formula. Flags: " + str(test_case.generate_run_argument_list(*executor.tool)))
+            except subprocess.TimeoutExpired:
+                return
+        # NOTE: We can't check for RC as some CORE test cases results in PARSING ERRORS :) 
         if TestCase.RUN_ONLY:
-            if rc != 0:
-                self.fail(f"Wrong output for process. Bombed out with exit code {rc}")
```

Comment:
What's the problem with the PARSING ERROR tests here? They did exist before, no?

Reply:
No problem, I am just not sure what we are verifying at this point. Are we verifying for a successful execution? It can be verification successful, SMTLIB, verification failed, parsing error.  We could try to validate every one of them individually, but then we don't know what the original test case wanted.

Thats why I was annoyed that the PARSING ERROR for a limitation is in core.  That being said, we could also consider testing whether the frontend can find issues correctly.

So, I preferred to just disable it. RUN_ONLY should check some byproduct of the run, not the run itself.